src/main/java/Loopy.java

```diff
@@ -0,0 +1,191 @@
+            tasks.add(new TodoTask(description));
+
+            System.out.println("Got it. I've added this task: ");
+            System.out.println(tasks.get(tasks.size() - 1));
+            System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+        //}
+    }
+    private static void addDeadline(String task) {
+       //separate the input into 2 substrings: description and deadline
+        int splitPosition = task.indexOf("/");
+        String description = task.substring(9, splitPosition);
+        String deadline = task.substring(splitPosition + 4, task.length());
+
+        tasks.add(new DeadlineTask(description,deadline));
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + tasks.get(tasks.size() - 1));
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    private static void addEvent(String task){
+        int fromPosition = task.indexOf("from");
+        int toPosition = task.indexOf("to");
+        String description = task.substring(6, fromPosition - 1);
+        String fromDate = task.substring(fromPosition + 5, toPosition - 1);
+        String toDate = task.substring(toPosition + 3, task.length());
+
+        tasks.add(new EventTask(description, fromDate, toDate));
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + tasks.get(tasks.size() - 1));
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+    private static void displayTaskList() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < tasks.size(); i++) {
+            Task currentTask = tasks.get(i);
+            String taskInfo = currentTask.toString();
+            System.out.println((i + 1) + ". " + taskInfo);
+        }
+    }
+
+    private static void markTaskAsDone(String task) {
+        int taskIndex = Integer.parseInt(task.substring(5)) - 1;
+        if (taskIndex >= 0 && taskIndex < tasks.size()) {
+            Task currentTask = tasks.get(taskIndex);
+            currentTask.markAsDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(" " + currentTask);
+        } else {
+            System.out.println("Invalid task index. Please provide a valid index.");
+        }
+    }
+
+    private static void markTaskAsUndone(String task) {
+        int taskIndex = Integer.parseInt(task.substring(7)) - 1;
+        if (taskIndex >= 0 && taskIndex < tasks.size()) {
+            Task currentTask = tasks.get(taskIndex);
+            currentTask.markAsNotDone();
+            System.out.println("OK, I've marked this task as not done yet:");
+            System.out.println(" " + currentTask);
+        } else {
+            System.out.println("Invalid task index. Please provide a valid index.");
+        }
+    }
+}
+
+class Task {
```

Comment:
Avoid creating all classes in the same file, separate them into different files.